src/app/services/news.service.ts

```diff
@@ -1,57 +1,54 @@
 import { Injectable } from '@angular/core';
 import { HttpClient } from '@angular/common/http';
-import { Observable, from } from 'rxjs';
+import { Subject } from 'rxjs';
 import { map } from 'rxjs/operators';
 import { News } from '../models/News';
+import { RegionSelectService } from './region-select.service';
+import { Region } from '../models/Region';
 
 export interface NewsService {
-  getNews(): Observable<News[]>;
+  getNewsObservable(): Subject<News[]>;
 }
 
 @Injectable({
   providedIn: 'root'
 })
-export class DummyNewsService implements NewsService {
-  getNews(): Observable<News[]>  {
-    return from([[
-      {
-        url: 'http://www.google.com',
-        title: 'COVID kills',
-        description: 'Dumb ppl still not wearing masks.'
-      },
-      {
-        url: 'http://www.bridgebase.com/v3',
-        title: 'Moron dies after injecting himself with bleach',
-        description: 'Trump claims injecting bleach as a COVID treatment is "very, very bad"'
+export class SmartableAiNewsService implements NewsService {
+  private url = 'https://api.smartable.ai/coronavirus/news/';
+  private subscriptionKey = '955eacf9525e45dd8d46a07b7daa649e';
+  private newsObserver: Subject<News[]>;
+
+  constructor(private httpClient: HttpClient, private regionService: RegionSelectService) {
+    this.newsObserver = new Subject<News[]>();
+    this.getSubscription(regionService.getRegion());
+    regionService.getRegionObservable().subscribe(
+      region => {
+        this.getSubscription(region);
       }
-    ]]);
+    );
   }
-}
 
-@Injectable({
-  providedIn: 'root'
-})
-export class ActualNewsService implements NewsService {
-  url = 'https://api.smartable.ai/coronavirus/news/';
-  subscriptionKey = '955eacf9525e45dd8d46a07b7daa649e';
-  region = 'global';
-
-  constructor(private httpClient: HttpClient) {}
+  getNewsObservable(): Subject<News[]> {
+    return this.newsObserver;
+  }
 
-  getNews(): Observable<News[]> {
-    return this.httpClient
+  private getSubscription(region: Region): void {
+    this.httpClient
       .get(
-        this.url + this.region,
+        this.url + region.code,
         {
           headers: { 'Subscription-Key': this.subscriptionKey }
         })
-      .pipe(map((response: {news: any}) =>
+      .pipe(map((response: { news: any }) =>
         response.news.map(
           ({ title, excerpt, webUrl }: { title: string, excerpt: string, webUrl: string }) => ({
-          title,
-          url: webUrl,
-          description: excerpt
-        }))
-      ));
+            title,
+            url: webUrl,
+            description: excerpt
```

Comment:
I think it might make sense to create interfaces/classes for each api response/object. Then that would shorten the pipe and make it clearer whats coming in/out?

check out [the way i did something similar in #19 ](https://github.com/SL-edi/covid-visualisation/blob/e4c9a3da33ac92b833ab923ff5591c5e7ce2516e/src/app/services/externalApis/covid-19-api.service.ts#L21-L28) to see what i mean. That way you wouldnt need to cast `news` as `any` or have all the type definitions in the `response.news.map`

plus if we've got a class for the resulting news item `({title, url, description})` then if we have a second source api for news or decide to change the format of our internal newsItem then we can just do it within the constructor of the newsItem class.

Reply:
I wrote it this way on purpose, because different apis will have in general different response structure, so instead of polluting the global namespace with a response object for each api, I think it's better to keep them contained within their respective classes.

I think I can define an interface inside a class though, so that may be a nicer way of doing this.

Reply:
oh the interface inside a class is a good idea, I think I'll switch over to using those too.

Reply:
You actually can't do that, but you can create an interface right next to the class and never export it.